src/mpid/ch4/netmod/ofi/ofi_impl.h

```diff
@@ -1064,16 +1064,16 @@ static int MPIDI_OFI_gpu_progress_task(MPIDI_OFI_gpu_task_t * gpu_queue[], int v
                 chunk_req->parent = request;
                 chunk_req->event_id = MPIDI_OFI_EVENT_SEND_GPU_PIPELINE;
                 chunk_req->buf = task->buf;
-                MPIDI_OFI_CALL_RETRY(fi_tsenddata
-                                     (MPIDI_OFI_global.ctx
-                                      [MPIDI_OFI_REQUEST(request, pipeline_info.ctx_idx)].tx,
-                                      task->buf, task->len, NULL /* desc */ ,
-                                      MPIDI_OFI_REQUEST(request, pipeline_info.cq_data),
-                                      MPIDI_OFI_REQUEST(request, pipeline_info.remote_addr),
-                                      MPIDI_OFI_REQUEST(request,
-                                                        pipeline_info.match_bits) |
-                                      MPIDI_OFI_GPU_PIPELINE_SEND, (void *) &chunk_req->context),
-                                     vni, fi_tsenddata);
+                MPIDI_OFI_CALL(fi_tsenddata
```

Comment:
Is this an intentional change? What was it fixing?

Reply:
Yes, I wanted to avoid the case that CALL_RETRY calls into progress engine which may do some re-entry.

Reply:
I see. Could you split the commits?

Reply:
done the splitting

Reply:
OK, so the idea here is that if this fails, the entry stays in the queue and will get issued at another time?

Reply:
yes, it should stay in the queue